test/unit/gateways/blue_snap_test.rb

```diff
@@ -543,6 +543,17 @@ def test_localizes_currencies
     end
   end
 
+  def test_optional_idempotency_key_header
+    response = stub_comms(@gateway, :raw_ssl_request) do
+      @gateway.purchase(@amount, @credit_card, @options.merge({ idempotency_key: 'test123' }))
+    end.check_request do |headers|
+      headers && headers['Idempotency-Key'] == 'test123'
```

Comment:
We could replace `'test123'` with any value and this test would still pass.

This is because we aren't asserting anything here. It needs to be `assert_equal 'test123', headers['Idempotency-Key']`.

Once you fix that you will see that the test fails because `check_request` hasn't been implemented correctly here. In this instance the headers end up being the fourth value returned by `check_request`, so the syntax should look like:
```
end.check_request do |_method, _url, _data, headers|
```
kind of like you see [here](https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/blue_snap_test.rb#L312).

In my experience, it is not easy to see how `check_request` works under the hood. It yields values depending on how the gateway adapter formulates the request, so whenever I need to use it to extract a value from a request I usually just declare all four values in the block and add a binding inside the block as well to look at what it has extracted.

Reply:
That makes sense. I knew something was a little off here. And thanks for the tip.